core/types/block.go

```diff
@@ -286,7 +288,9 @@ func NewBlock(header *Header, body *Body, receipts []*Receipt, hasher TrieHasher
 		b.header.Bloom = CreateBloom(receipts)
 	}
 
-	if len(uncles) == 0 {
+	// We prevent setting the EmptyUncleHash only when we are on a migrated
+	// chain and the block is pre-gingerbread.
+	if len(uncles) == 0 && !(bType.IsMigratedChain() && !bType.IsGingerbread(header.Number)) {
```

Comment:
I think this new logic prevents setting the EmptyUncleHash only when on a migrated chain and the block is post gingerbread

Reply:
Oh sorry, think I missed the parenthesis